### ISSUE TITLE
wer on test dataset is differeent 

### ISSUE THREAD

**Issue by guoying1030** [fchollet/keras#6172]
When I use the distributed tensorflow and keras combined training, each epoch will be saved on the worker0 model for the keras model, but on the validation set, before saving the model, its wer is 0.17, when saved as keras model, I use a separate program to load keras model, using the same proof set, why its wer becomes 0.22
